soundcard/pulseaudio.py

```diff
@@ -371,7 +361,63 @@ def record(self, numframes):
                     captured_frames += len(chunk)/self.channels
             else:
                 time.sleep(0.001)
-        return numpy.reshape(numpy.concatenate(captured_data), [-1, self.channels])
+        return captured_data, captured_frames, len(chunk)
+
+    def record_chunk(self, numframes):
```

Comment:
I think I have made up my mind now, and would prefer this function to not exist at all.